### ISSUE TITLE
issue import modelgrid from gridgen

### ISSUE THREAD

**Issue by MichelCraninx** [modflowpy/flopy#2364]
Hello
I have set-up a quadtree model with gridgen (disv-object). Because this takes a long time to calculate I would like to save the modelgrid and import it. For this I have run a simulation so I can import the disv-object and model. When I import the modelgrid the spatial data is missing when I print mg (mg = gwf.modelgrid). However when I plot the modelgrid (mg.plot()), the modelgrid is displayed with the correct coordinates? When I follow the steps: https://flopy.readthedocs.io/en/latest/Notebooks/modelgrid_examples.html, I can change the coordinates to the correct coordinates, however when I plot the new modelgrid the modelgrid is not on the correct position. How can I correct load the modelgrid, resulting from the gridgen calculations? 

![image](https://github.com/user-attachments/assets/caf69f82-9ecd-48c4-836f-da2a49011126)
![fig_modelgrid_error](https://github.com/user-attachments/assets/d78ad737-2d78-4baf-aef0-58481d7f36a2)

**Comment by martclanor** [comment]
Oh okay, I'm not sure, but can you check if this is also the case before you exported the modelgrid? Just to rule-out that this has something to do with the re-importing of the modelgrid.

I just tested this but on a different grid:
![image](https://github.com/user-attachments/assets/6c7389b2-f1d8-47d1-9d42-7f2cd2b03ccb)

And then setting coord info seems to work fine:
![image](https://github.com/user-attachments/assets/1b1ff082-bd21-4b29-835f-69f19125686f)

Btw, I am using flopy version 3.8.2 here:
```python
flopy.__version__  # '3.8.2'
```

**Comment by MichelCraninx** [comment]
This is how it looks after the import modelgrid from sim: 
![image](https://github.com/user-attachments/assets/9f198216-b714-4cde-9670-fd1164d235d5)
This is how it looks before importing (original) 
![image](https://github.com/user-attachments/assets/84647e9e-4a4b-4f55-b313-290fb2096165)
The goal is to export the 'base_grid' after gridgen to use for further steps, for example resampling raster data, it works when I use the 'base_grid' directly coming from the gridgen step: 
![image](https://github.com/user-attachments/assets/0ac23ca5-2e22-4e96-ba05-9b04450d3ee9)
But when I use the modelgrid after importing for further steps: for example resample rasters 
![image](https://github.com/user-attachments/assets/ad829d61-7f45-4e36-a07f-5a0f696c5b26)
then I get this error when I plot the results: 
![image](https://github.com/user-attachments/assets/430d3553-c302-416b-952a-28be39573fdf)
![image](https://github.com/user-attachments/assets/428c58e1-d2a5-4de4-80bb-a1b565cb5974)
This is not the case when working with the original base_grid ...
![image](https://github.com/user-attachments/assets/1f5424fc-c8e3-4e62-9d8d-4897a5456ac3)
It is not clear where the error is, but it seems that importing the modelgrid is causing the error?

**Comment by martclanor** [comment]
I just realized that your "imported modelgrid" is the one coming from the `gwf` object which is not aware of the `crs` because the grid is defined via the `ModflowGwfdisv` class (which doesn't have the `crs` parameter). 

Does it work if you do something like this upon importing the modelgrid?:

```python
mg_test = gwf_quadtree_test.modelgrid
mg_test.set_coord_info(crs="EPSG:31370")
```

**Comment by MichelCraninx** [comment]
Hello, no, I get still the same error, when I only adjust the crs, the plot of the mg_test seems on the wright position, but the print gives = 0, and I cannot do the resampling, with error unsupported operand type, when I both correct coordinates and crs, the plot is on the wrong location and I get the error Raster and modelgrid do not intersect.

**Comment by martclanor** [comment]
Oh okay. I'm starting to run out of ideas 😅. 

I'm wondering why printing the `VertexGrid` object in both cases show that xoffset (xll) and yoffset (yll) are zero. Did you also pass the correct `xorigin` and `yorigin` in your `ModflowGwfdisv` object?

**Comment by jlarsen-usgs** [comment]
@martclanor and @MichelCraninx 

When gridgen builds a VertexGrid from a StructuredGrid that has offsets applied to it (in FloPy_, gridgen will calculate the vertices for the new VertexGrid from the offset original grid. Therefore, the VertexGrid's vertices are specified in the correct place and no offsets need to be applied to the new grid. Hense, xoff=0, yoff=0, angrot=0 when you load the new modelgrid. If you then apply the offsets to it, you are translating vertices that are in the correct place by whatever offset distance you applied. 

In short, xoffset and yoffset should both equal 0 on your new grid (do not apply offsets to it) as the actual vertices that are defined for each cell correspond to the geographic coordinates of the StructuredGrid it was created from.

**Comment by MichelCraninx** [comment]
Thank you very much martclanor and jlarsen! So I have to find a manner to load correctly the base_grid after gridgen. When I print the properties of the grid after import from the sim object, it is a "vertexgrid" with fine and rough resolutions, as constructed during with gridgen. However, I cannot explain the errors when I use the modelgrid for resampling for example: "unsupported operand type". It is already good to know that is not related with the coordinates.

**Comment by jlarsen-usgs** [comment]
@MichelCraninx 
It's hard to say what's going on without seeing the full error message here. Which specific function call is leading to the ValueError? What's the `type()` of the `rA0220_thickness_clip`?

**Comment by MichelCraninx** [comment]
Hi
I make a modelgrid with gridgen and couple the base_grid to the disv object:
![image](https://github.com/user-attachments/assets/8ff92275-7bfc-414c-bbc3-e6aa8dfbee4b)
For saving the modelgrid I start a simulation and load the results in a sim:
![image](https://github.com/user-attachments/assets/dfaafda0-4b99-4415-98c4-30aa93992f60)
The modelgrid looks ok:
![image](https://github.com/user-attachments/assets/8d126a73-0fca-4e02-8591-355b83467b8c)
![image](https://github.com/user-attachments/assets/44ddc55e-0515-47ce-88c5-dba370353c1b)
Then I do a resampling with this modelgrid:
![image](https://github.com/user-attachments/assets/b51248db-8bb4-45dc-822c-9dc422c2a836)
![image](https://github.com/user-attachments/assets/8cb7e914-d6e5-4b42-b921-91886d5c103b)
When I plot the resampled plot I get this error:
![image](https://github.com/user-attachments/assets/bf3ba4c6-bec2-411f-bc19-3a30dbf0f8e9)
![image](https://github.com/user-attachments/assets/7598a671-4169-48fe-92a5-94c1f35ddf5b)

In case I work directly from the base_grid after gridgen it is ok: 
![image](https://github.com/user-attachments/assets/dd369572-ba9b-4730-8510-ddecefaf7e2c)

I hope that there is a way I can import the modelgrid after gridgen because we would like to increase the resolution, which will take some time.

Thank you very much!
Michel

**Comment by jlarsen-usgs** [comment]
I see what's going on. Your `base_grid` doesn't have the `nlay` parameter set for it. Here is one way to fix it in your code:

```python
if local_grid_option == 1:
    gridprops = g.get_gridprops_vertexgrid()
    base_grid = flopy.discretization.VertexGrid(nlay=1, **gridprops)
```

Also `rlayer_top` is a numpy array that is resampled from your raster. As such, it will not have a `head()` method. Only `flopy.mf6.ModflowGwf` and `flopy.mf6.ModflowGwfoc` objects have the `head()` method. Example:

```python
gwf = sim.get_model("my model name")
head_obj = gwf.head()
head_array_data = head_obj.get_alldata()  # 4d array of (time, nlay, nrow, ncol)
```
